pandas/core/window/rolling.py

```diff
@@ -1884,6 +1884,9 @@ def _validate(self):
             else:
                 self._win_freq_i8 = freq.nanos
 
+            if self._on.dtype == "M8[us]":
+                self._win_freq_i8 /= 1000
+
```

Comment:
@jbrockmendel to generalize this, we could easily add two more `elif dtype == "M8[ms/s]"` checks and do this division explicitly for each of the three non-nanoseconds resolutions, but do you know of a better pattern? Is there some helper function that could be used here?

Or otherwise another way to convert the `freq` to the correct resolution instead of calling `freq.nanos`?

Reply:
maybe `Timedelta(freq).as_unit(correct_unit)._value`